BLD/BUG: Exclusions path separator for Windows

ChadFulton opened the pull request:
Closes #3499

Makes sure the exclusions file has the appropriate separator for comparison in `cythonize.py`. Tested on my Windows machine and works there.

josef-pkt commented:
You could rebase on master after we merge #3511, then we see that the fix works.
Currently appveyor shows the cythonize problem of #3499

ChadFulton commented:
Rebased.

ChadFulton commented:
This build fixes the scipy 0.14 Appveyor (the failure is the usual `test_simulate.py` failure). We can probably merge this.

josef-pkt commented:
green with scipy 0.14, red is the usual simulate problem

merging

josef-pkt commented:
(simultaneous edit, but I was faster hitting the green button  :)

Thanks, it's good to have the test for the conditional compilation also for this case

bashtage commented:
Would be safer to use 

```python
os.path.join('statsmodels','tsa','kalmanf','kalman_loglike.c')
```

ChadFulton commented:
> Would be safer to use `os.path.join('statsmodels','tsa','kalmanf','kalman_loglike.c')`

I guess so; since the compilation part itself seems to work I was afraid of breaking something new if I changed it, so I just changed the part that was breaking.
